Lib/fractions.py

```diff
@@ -832,15 +840,15 @@ def __rpow__(b, a):
 
     def __pos__(a):
         """+a: Coerces a subclass instance to Fraction"""
-        return Fraction(a._numerator, a._denominator, _normalize=False)
+        return Fraction._from_pair(a._numerator, a._denominator)
 
     def __neg__(a):
         """-a"""
-        return Fraction(-a._numerator, a._denominator, _normalize=False)
+        return Fraction._from_pair(-a._numerator, a._denominator)
 
     def __abs__(a):
         """abs(a)"""
-        return Fraction(abs(a._numerator), a._denominator, _normalize=False)
+        return Fraction._from_pair(abs(a._numerator), a._denominator)
```

Comment:
If Fraction is really immutable we could optimize here by returning self in case `a._numerator` is positive. But i am not sure whether the gain is large enough for the added code complexity.

Reply:
> If Fraction is really immutable 

"[Fraction](https://docs.python.org/3/library/fractions.html?highlight=fraction#fractions.Fraction) instances are hashable, and should be treated as immutable." (c) docs

On the ~maste~main:
```
$ ./python -m timeit -s 'from fractions import Fraction as Q; a=Q(11,5)' 'abs(a)'
50000 loops, best of 5: 4.98 usec per loop
```
With
```
diff --git a/Lib/fractions.py b/Lib/fractions.py
index 49a3f2841a..53086e7841 100644
--- a/Lib/fractions.py
+++ b/Lib/fractions.py
@@ -840,6 +840,8 @@ def __neg__(a):
 
     def __abs__(a):
         """abs(a)"""
+        if a._numerator >= 0:
+            return a
         return Fraction(abs(a._numerator), a._denominator, _normalize=False)
 
     def __int__(a, _index=operator.index):
```
I see:
```
$ ./python -m timeit -s 'from fractions import Fraction as Q; a=Q(11,5)' 'abs(a)'
1000000 loops, best of 5: 277 nsec per loop
```
On this pr:
```
$ ./python -m timeit -s 'from fractions import Fraction as Q; a=Q(11,5)' 'abs(a)'
100000 loops, best of 5: 2.71 usec per loop
```
So, there could be some speed bonus...  Ditto for `__neg__`.

For `__pos__` we, probably, could return self. Not sure which use case expects the type casting to the Fraction in this method...

Reply:
> If Fraction is really immutable we could optimize here by returning self in case `a._numerator` is positive.

That would be a behaviour change for instances of `Fraction` subclasses: currently, `abs` on an instance of a strict `Fraction` subclass will return something of exact type `Fraction`.